Code/Tools/ProjectManager/Source/PythonBindings.cpp

```diff
@@ -1051,12 +1062,13 @@ namespace O3DE::ProjectManager
         return result && refreshResult;
     }
 
-    bool PythonBindings::AddGemRepo(const QString& repoUri)
+    AZ::Outcome<void, AZStd::string> PythonBindings::AddGemRepo(const QString& repoUri)
     {
         bool registrationResult = false;
         bool result = ExecuteWithLock(
             [&]
             {
+                RedirectOutput::lastPythonError.clear();
```

Comment:
I'd move this into ExecuteWithLockErrorHandling() so you don't need to remember to call it in every high level API function